applications/luci-app-filemanager/htdocs/luci-static/resources/view/system/filemanager.js

```diff
@@ -0,0 +1,3616 @@
+		}
+		if (inputs.currentDirectory) {
+			inputs.currentDirectory.value = config.currentDirectory || '/';
+		}
+		if (inputs.windowWidth) {
+			inputs.windowWidth.value = config.windowSizes.width;
+		}
+		if (inputs.windowHeight) {
+			inputs.windowHeight.value = config.windowSizes.height;
+		}
+		if (inputs.editorTextWidth) {
+			inputs.editorTextWidth.value = config.editorContainerSizes.text.width;
+		}
+		if (inputs.editorTextHeight) {
+			inputs.editorTextHeight.value = config.editorContainerSizes.text.height;
+		}
+		if (inputs.editorHexWidth) {
+			inputs.editorHexWidth.value = config.editorContainerSizes.hex.width;
+		}
+		if (inputs.editorHexHeight) {
+			inputs.editorHexHeight.value = config.editorContainerSizes.hex.height;
+		}
+	},
+
+	renderEditor: function(filePath) {
+		var self = this;
+
+		var editorContainer = document.getElementById('editor-container');
+
+		// Clear the editor container
+		editorContainer.innerHTML = '';
+
+		// Get the sizes from the config
+		var mode = self.editorMode; // 'text' or 'hex'
+		var editorSizes = config.editorContainerSizes[mode] || {
+			width: 850,
+			height: 550
+		};
+
+		// Create the editor content container
+		var editorContentContainer = E('div', {
+			'class': 'editor-content',
+			'style': 'flex: 1; display: flex; overflow: hidden;'
+		}, []);
+
+		// Action buttons array
+		var actionButtons = [];
+
+		if (mode === 'text') {
+			// Create line numbers div (initially hidden)
+			var lineNumbersDiv = E('div', {
+				'id': 'line-numbers',
+				'class': 'line-numbers',
+				'style': 'display: none;' // Initially hidden
+			}, []);
+
+			// Create textarea for text editing
+			var editorTextarea = E('textarea', {
+				'wrap': 'off',
+				'id': 'editor-textarea',
+				'style': 'flex: 1; resize: none; border: none; padding: 0; margin: 0; overflow: auto;'
+			}, [self.fileContent || '']);
+
+			// Append line numbers and textarea to the editor content container
+			editorContentContainer.appendChild(lineNumbersDiv);
+			editorContentContainer.appendChild(editorTextarea);
+
+			// Add event listeners for updating line numbers and synchronizing scroll
+			editorTextarea.addEventListener('input', self.updateLineNumbers.bind(self));
+			editorTextarea.addEventListener('scroll', self.syncScroll.bind(self));
+
+			// Define action buttons specific to Text Mode
+			actionButtons = [
+				E('button', {
+					'class': 'btn cbi-button-save custom-save-button',
+					'click': function() {
+						self.handleSaveFile(filePath);
+					}
+				}, _('Save')),
+				E('button', {
+					'class': 'btn',
+					'id': 'toggle-line-numbers',
+					'style': 'margin-left: 10px;',
+					'click': function() {
+						self.toggleLineNumbers();
+					}
+				}, _('Toggle Line Numbers')),
+				E('button', {
+					'class': 'btn',
+					'id': 'toggle-hex-mode',
+					'style': 'margin-left: 10px;',
+					'click': function() {
+						self.toggleHexMode(filePath);
+					}
+				}, _('Toggle to Hex Mode'))
```

Comment:
Reorder these so that toggle of hex/text is consistent

Reply:
Done